arbos/block_processor.go

```diff
@@ -168,6 +170,7 @@ func ProduceBlockAdvanced(
 	txes types.Transactions,
 	delayedMessagesRead uint64,
 	lastBlockHeader *types.Header,
+	lastHotShotHeader *espresso.Header,
```

Comment:
1. `lastHotShotHeader` is added as an input to this method, which is the bulk of the "State Transition Function". Now, the mock namespace proof can be verified against the provided header (below I compare the provided preimage header to the header in the justification, but I guess we don't necessarily need the header in the justification at all if we do things this way).

Reply:
Re: not putting the header in the justification: we could do that, but it would kind of break the "L1 DA" property, where everything you need to validate the L2 chain is available on L1.

If we were to do this, we could probably remove the justification completely and just have the validator fetch whatever proof it needs from HotShot and provide it as a hint to the WASM verifier.

Reply:
Yes, we assume 1:1 here, last probably isn't the best word, it's more like the HotShotSafeHead